push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/AdmPushNotificationSender.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.sender;
+
+
+import org.jboss.aerogear.unifiedpush.api.AdmVariant;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+import org.jboss.aerogear.unifiedpush.utils.AeroGearLogger;
+import org.jboss.aerogear.adm.ADM;
+import org.jboss.aerogear.adm.AdmService;
+import org.jboss.aerogear.adm.PayloadBuilder;
+
+import java.util.Collection;
+import java.util.Set;
+
+@SenderType(AdmVariant.class)
+public class AdmPushNotificationSender implements PushNotificationSender {
+    private final AeroGearLogger logger = AeroGearLogger.getInstance(AdmPushNotificationSender.class);
+
+    @Override
+    public void sendPushMessage(Variant variant, Collection<String> clientIdentifiers, UnifiedPushMessage pushMessage, NotificationSenderCallback senderCallback) {
+        final AdmService admService = ADM.newService();
+
+        final PayloadBuilder builder = ADM.newPayload();
+
+        //flatten the "special keys"
+        builder.dataField("alert", pushMessage.getMessage().getAlert());
+        builder.dataField("sound", pushMessage.getMessage().getSound());
+        builder.dataField("badge", "" + pushMessage.getMessage().getBadge());
+
+        // if present, apply the time-to-live metadata:
+        int ttl = pushMessage.getConfig().getTimeToLive();
+        if (ttl != -1) {
+            builder.expiresAfter(ttl);
+        }
+
+        //dirty hack for cordova,
+        //TODO should be removed once we have our clients SDKs, tracked by AGPUSH-1269
+        builder.dataField("message","useless payload");
+
+        //Handle consolidation key
+        builder.consolidationKey("my other app");
```

Comment:
I don't understand the ""my other app"" value

On Mon, Feb 9, 2015 at 3:40 PM, Sebastien Blanc <notifications@github.com>
wrote:

> In
> push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/AdmPushNotificationSender.java
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/480#discussion_r24331608>
> :
>
> > +        builder.dataField("alert", pushMessage.getMessage().getAlert());
> > +        builder.dataField("sound", pushMessage.getMessage().getSound());
> > +        builder.dataField("badge", "" + pushMessage.getMessage().getBadge());
> > +
> > +        // if present, apply the time-to-live metadata:
> > +        int ttl = pushMessage.getConfig().getTimeToLive();
> > +        if (ttl != -1) {
> > +            builder.expiresAfter(ttl);
> > +        }
> > +
> > +        //dirty hack for cordova,
> > +        //TODO should be removed once we have our clients SDKs, tracked by AGPUSH-1269
> > +        builder.dataField("message","useless payload");
> > +
> > +        //Handle consolidation key
> > +        builder.consolidationKey("my other app");
>
> is used for their "sync" facilities, it's documented in the Message class
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/480/files#r24331608>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf